sample/Kmd.Logic.CitizenDocuments.Client.Sample/Program.cs

```diff
@@ -59,44 +58,45 @@ private static async Task Run(AppConfiguration configuration)
                 };
                 configuration.Citizen.SubscriptionId = configuration.SubscriptionId;
                 configuration.Citizen.Serviceuri = configuration.Serviceuri;
                 var tokenProviderFactory = new LogicTokenProviderFactory(tokenProviderOptions);
                 var citizenDocumentClient = new CitizenDocumentsClient(httpClient, tokenProviderFactory, configuration.Citizen);
-                var uploadDocument = await citizenDocumentClient.UploadAttachmentWithHttpMessagesAsync(configuration.ConfiguartionId, configuration.RetentionPeriodInDays, configuration.Cpr, configuration.DocumentType, configuration.Document, configuration.DocumentName).ConfigureAwait(false);
+                var uploadDocument = await citizenDocumentClient.UploadAttachmentWithHttpMessagesAsync(configuration.ConfigurationId, configuration.RetentionPeriodInDays, configuration.Cpr, configuration.DocumentType, configuration.Document, configuration.DocumentName).ConfigureAwait(false);
 
                 if (uploadDocument == null)
                 {
-                    Log.Error("There is error occured in upload");
+                    Log.Error("An error occurred while document upload operation");
                     return;
                 }
-                Log.Information($"Document uploaded successfully and details are :-  DocumentId : {uploadDocument.DocumentId} ; DocumentType : {uploadDocument.DocumentType} ; FileAccessPageUrl : {uploadDocument.FileAccessPageUrl} ");
+
+                Log.Information("Document uploaded successfully and details are :-  DocumentId : {DocumentId} ; DocumentType : {DocumentType} ; FileAccessPageUrl : {FileAccessPageUrl}", uploadDocument.DocumentId, uploadDocument.DocumentType, uploadDocument.FileAccessPageUrl);
 
                 var sendDocument = await citizenDocumentClient.SendDocumentWithHttpMessagesAsync(new SendCitizenDocumentRequest
                 {
-                    ConfigurationId = new Guid(configuration.ConfiguartionId),
+                    ConfigurationId = new Guid(configuration.ConfigurationId),
                     SendingSystem = configuration.SendingSystem,
                     Cpr = configuration.Cpr,
                     DocumentType = configuration.SendDocumentType,
                     CitizenDocumentId = uploadDocument.DocumentId,
-                    Title = configuration.title,
+                    Title = configuration.Title,
                     DigitalPostCoverLetterId = uploadDocument.DocumentId,
-                    SnailMailCoverLetterId = uploadDocument.DocumentId
-                });
+                    SnailMailCoverLetterId = uploadDocument.DocumentId,
+                }).ConfigureAwait(false);
+
                 if (sendDocument == null)
                 {
-                    Log.Error("There is error occured in send document");
+                    Log.Error("An error occurred while send document operation");
```

Comment:
bad grammar. I suggest something like:

> An unknown error occurred while sending the document

Furthermore, why should this return null? Isn't null a bad response, since it's very difficult for end users to understand what it means, and they have to explicitly check for it?

Reply:
Corrected as per your input

Reply:
I didn't get a response to the question about returning null

Reply:
The thought behind returning null was:
This is a sample project and whoever gonna use the nuget will be implementing in their own way. This sample is just for the reference on how to use the nuget. So we just thought, returning null or string shouldn't matter much in the sample

Reply:
I am not concerned with the sample, but rather the issue that has been observed while reviewing this sample is this:

the method `SendDocumentWithHttpMessagesAsync` returns `null`. If it returned something better than null, the sample would improve, but more importantly, the consumer of the API would have some information easily available to decide what to do next.

Reply:
Yeah sure, I see that and makes sense to return proper message. Fixed it.